File: c/30.c
#include <stdio.h>
#include <math.h>

int solution(int n)
{
	int orig = n, sum = 0;
	do sum += pow(n % 10, 5); while (n /= 10);
	return sum == orig;
}

int main(void)
{
	int i, sum = 0;

	for (i = 2; i < 200000; ++i)
		sum += solution(i) ? i : 0;

	printf("%d\n", sum);
	return 0;
}

File: js/30.js
function solution(n) {
    return n.toString().split('').map(function(c) {
        return Math.pow(c * 1, 5);
    }).reduce(function(a, b) {
        return a + b;
    }) == n;
}

sum = 0
for (var i = 2; i < 200000; ++i)
    sum += solution(i) ? i : 0;

console.log(sum);
